pandas/core/base.py

```diff
@@ -7,8 +7,10 @@
 from pandas.core import common as com
 import pandas.core.nanops as nanops
 import pandas.lib as lib
+from pandas.compat.func_defaults import TRANSPOSE_DEFAULTS
```

Comment:
In numpy_compat

```
def transpose(obj, *args, **kwargs):
    validate_args_and_kwargs('transpose', args, kwargs, 0,
                                 TRANSPOSE_DEFAULTS)
```

I purpose is to make ALL validation a 1-liner in the main code body to avoid cluttering with unecessary things, and all of the messy compat stuff go in ``numpy_compat.py``. Simply move all of the code there.